### ISSUE TITLE
Not seeing expected performance gain in dual GPU training

### ISSUE THREAD

**Issue by vijaygill** [tensorpack/tensorpack#1435]
Hi,

I am trying to use two GPU's (RTX 2080 Super) for training. They work but the expected performance gain is not there.

If I run training of a model (MasterRCNN) on any one of them, I get about 4.8-4.9 samples/sec through put. But when I use both (along with BACKBONE.NORM = "GN" and FPN.NORM = "GN" and BACKBONE.WEIGHTS = "/training/COCO-MaskRCNN-R50FPN2xGN.npz") the through put is like 5.2-5.5 samples/sec.

I have not been able to train for more than a few epochs anyway because of weights exploding (I start getting total loss NaN after a few epochs no matter how low I set the TRAIN.BASE_LR. I set it to even 1e-6 and still got the issue.

Even though I may not end up using multi-GPU due to the NaN issue, is there something I should be looking at for getting the performance gain?

### 1. What you did:

Ran FaterRCNN training using two GPUs. Along with group-normalization and suggested base weights. I am using my custom dataset.

### 2. What you observed:
I observed the gain in performance was less than expected.
[Log file is attached here](https://github.com/tensorpack/tensorpack/files/4597577/log.log). Note I forgot to change the BASE_LR hence I got NaN quite early, but I think it is not relevant to this issue.

### 3. What you expected, if not obvious.
I expected higher throughput than what I witnessed.

### 4. Your environment:
--------------------  -------------------------------------------------------------------
sys.platform          linux
Python                3.6.9 (default, Nov  7 2019, 10:44:02) [GCC 8.3.0]
Tensorpack            0.10.1
Numpy                 1.18.4
TensorFlow            1.15.2/v1.15.0-92-g5d80e1e
TF Compiler Version   7.3.1 20180303
TF CUDA support       True
TF MKL support        False
TF XLA support        False
Nvidia Driver         /usr/lib/x86_64-linux-gnu/libnvidia-ml.so.440.82
CUDA                  /usr/local/cuda-10.0/targets/x86_64-linux/lib/libcudart.so.10.0.130
CUDNN                 /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.2
NCCL
CUDA_VISIBLE_DEVICES  Unspecified
GPU 0,1               GeForce RTX 2080 SUPER
Free RAM              59.64/60.95 GB
CPU Count             8
cv2                   4.2.0
msgpack               1.0.0
python-prctl          True
--------------------  -------------------------------------------------------------------

**Comment by ppwwyyxx** [comment]
As https://github.com/tensorpack/tensorpack/blob/master/examples/FasterRCNN/NOTES.md#efficiency says, the speed for first 10k iterations is expected to be slow it does not get better with more GPUs.

Using `export TF_CUDNN_USE_AUTOTUNE=0` can skip this warmup period.

**Comment by vijaygill** [comment]
@ppwwyyxx Thanks again for such a quick reply!👍👍👍


I will try as suggested.

**Comment by vijaygill** [comment]
@ppwwyyxx Just looked at the tensorboard dashboard and I noticed no improvement.

**Comment by ppwwyyxx** [comment]
Including the details following the [issue template](https://github.com/tensorpack/tensorpack/blob/master/.github/ISSUE_TEMPLATE/unexpected-problems---bugs.md) will give more information.

**Comment by vijaygill** [comment]
@ppwwyyxx Thanks for keeping an eye on this. I have updated original post with more information..

**Comment by ppwwyyxx** [comment]
I can't see anything obviously wrong from the logs.
The code does scale well to more GPUs when running with the machine&dataset&configuration in the documentation. Since all of them are different in your training it's difficult to tell what's the reason.

**Comment by vijaygill** [comment]
@ppwwyyxx I understand. Thanks for spending time on this though. I will close this issue.
